lib/engine/game/g_18_gb/game.rb

```diff
@@ -720,7 +720,7 @@ def legal_tile_rotation?(_entity, _hex, tile)
         def route_trains(entity)
           return super unless insolvent?(entity)
 
-          [@depot.min_depot_train]
+          @depot.upcoming.any? ? [@depot.min_depot_train] : []
```

Comment:
`Array.any?` (and `Array.none?`) shouldn't be used without a code block testing the array elements. To test whether an array has any elements you should use `Array.empty?`.
```suggestion
          @depot.upcoming.empty? ? [] : [@depot.min_depot_train]
```